syntax = "proto3";

message SetupRequest{
  string token = 1;
}

enum ServiceStatus {
  TRAINING = 0;
  ACTIVE = 1;
  IDLE = 2;
}

message SetupResponse {
  ServiceStatus status = 1;
}

message ImportRequest {
  repeated bytes files = 1;
}

enum Status {
  FAIL = 0;
  SUCCESSFUL = 1;
}

message ImportResponse {
  Status status = 1;
}

enum CommandType {
  PREDICT = 0;
  DOWNLOAD_CKPT = 1;
  TRAIN = 2;
  STOP_TRAINING = 3;
}

message CommandRequest {
  CommandType command = 1;
}

message CommandResponse {
  Status status = 1;
}

service Management{
  // Store and monitor the processing status of machine learning (ML), including active, training, etc.
  rpc Setup(SetupRequest) returns (SetupResponse);
  // Upload raw data into the system, allowing multiple files uploads simultaneously
  rpc Import(ImportRequest) returns (ImportResponse);
  // Execute an action corresponding to the command type in the request
  rpc Commands(CommandRequest) returns (CommandResponse);
}